File: SrdDatabase/Models/Quotas/Quota.cs
namespace SrdDatabase.Models.Quotas
{
    public class Quota
    {
        public int Id { get; }

        public int AmountPerYear { get; }

        public int StartYear { get; }

        public int? EndYear { get; }

        public int CongregationId { get; }

        public string Congregation { get; }

        public int ParishId { get; }

        public string Parish { get; }

        public int ArchdeaconryId { get; }

        public string Archdeaconry { get; }

        public Quota(
            int id,
            int amountPerYear,
            int startYear,
            int endYear,
            int congregationId,
            string congregation,
            int parishId,
            string parish,
            int archdeaconryId,
            string archdeaconry)
        {
            Id = id;
            AmountPerYear = amountPerYear;
            StartYear = startYear;
            EndYear = endYear;
            CongregationId = congregationId;
            Congregation = congregation;
            ParishId = parishId;
            Parish = parish;
            ArchdeaconryId = archdeaconryId;
            Archdeaconry = archdeaconry;
        }

        // for Dapper
        public Quota()
        {
        }
    }
}

File: SrdDatabase/ClientApp/src/models/quota.ts
import { PagedParameters, PagedResults } from "./shared";

export interface Quota {
    id?: number;
    amountPerYear?: number;
    congregationId?: number;
    congregation?: string;
    parishId?: number;
    parish?: string;
    archdeaconryId?: number;
    archdeaconry?: string;
    startYear?: number;
    endYear?: number;
    createdBy?: number;
}

export interface QuotaDetails {
    quota: Quota;
}

export interface QuotaParameters extends PagedParameters {
    parishId?: number;
    archdeaconryId?: number;
    congregationId?: number;
    startYear?: number;
    endYear?: number;
}

export interface QuotaResults extends PagedResults {
    quotas: Quota[];
}
